abb_irb1600_support/urdf/irb1600_6_12_macro.xacro

```diff
@@ -0,0 +1,173 @@
+    </collision>
+  </link>
+  <link name="${prefix}link_2">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600_6_12/visual/link_2.stl"/>
+      </geometry>
+      <material name="abb_orange"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600_6_12/collision/link_2.stl"/>
+      </geometry>
+      <material name="yellow"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_3">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/visual/link_3.stl"/>
+      </geometry>
+      <material name="abb_orange"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/collision/link_3.stl"/>
+      </geometry>
+      <material name="yellow"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_4">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/visual/link_4.stl"/>
+      </geometry>
+      <material name="abb_orange"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/collision/link_4.stl"/>
+      </geometry>
+      <material name="yellow"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_5">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/visual/link_5.stl"/>
+      </geometry>
+      <material name="abb_orange"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/collision/link_5.stl"/>
+      </geometry>
+      <material name="yellow"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_6">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/visual/link_6.stl"/>
+      </geometry>
+      <material name="abb_orange"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://abb_irb1600_support/meshes/irb1600/collision/link_6.stl"/>
+      </geometry>
+      <material name="yellow"/>
+    </collision>
+  </link>
+  <link name="${prefix}tool0"/>
+  <!-- end of link list -->
+  <!-- joint list -->
+  <joint name="${prefix}joint_1" type="revolute">
+    <origin xyz="0 0 0" rpy="0 0 0"/>
+    <parent link="${prefix}base_link"/>
+    <child link="${prefix}link_1"/>
+    <axis xyz="0 0 1"/>
+    <limit lower="-3.1416" upper="3.1416" effort="0" velocity="2.618"/>
```

Comment:
It might be better to use `${pi}` here: `3.1416` is a ceiled version of `pi`, and I've actually had robots run into their joint limits and faulting because MoveIt used that fourth decimal in some trajectories.

Alternatively: we could use the joint limits in degrees and use `${radians(..)}`.